### PR TITLE
Characters get update

### PR DESCRIPTION
**Pull request by Dragonfrut**
get all of the users characters

### CODE REVIEW

File: WildShape-Sheets-API/Services/PlayerCharacterService.cs [hunk 1]
```diff
@@ -22,6 +22,11 @@ public PlayerCharacterService(IConfiguration configuration, DataBaseService data
...
         public List<PlayerCharacter> GetPlayerCharacters() => _dataBaseService.playerCharacterCollection.Find(pc => true).ToList();
 
+        public List<PlayerCharacter> GetUsersPlayerCharacters(string email) {
+            var user = _userService.GetUserByEmail(email);
+            return user.Characters;
+        }
+
         public PlayerCharacter GetPlayerCharacterById(string id) => _dataBaseService.playerCharacterCollection.Find(pc => pc.Id == id).FirstOrDefault();
 
         public PlayerCharacter? CreatePlayerCharacter(string userEmail, PlayerCharacter pc)
```
Comment (smashingmouse) [inline WildShape-Sheets-API/Services/PlayerCharacterService.cs:32]: Pro tip:
`if (user.Characters?.Count <= 4) {`
You don't ever want a hard coded number. Always use a defned constant. MAX_NUMBER_PLAYER_CHARACTERS=4 (or something like that. That way it is clear what 4 is, and it is simple to change all occurances in the code rather that searching for "4" and getting who knows else what.

Comment (Dragonfrut) [reply WildShape-Sheets-API/Services/PlayerCharacterService.cs:32]: i did know that, it was in my brain but i forgot to do it

File: WildShape-Sheets-API/Services/PlayerCharacterService.cs [hunk 2]
```diff
@@ -48,5 +53,7 @@ public PlayerCharacterService(IConfiguration configuration, DataBaseService data
             _dataBaseService.playerCharacterCollection.DeleteOne(pc => pc.Id == pc.Id);
 
         }
+
```
Comment (smashingmouse) [inline WildShape-Sheets-API/Services/PlayerCharacterService.cs:56]: ChatGPT says:
The code you provided contains a small mistake in the DeletePlayerCharacter method. The issue is in the last line, where you are using pc => pc.Id == pc.Id as the filter for deleting the PlayerCharacter from the playerCharacterCollection. This is redundant, and you should use the correct filter instead.

Here's the corrected version of the DeletePlayerCharacter method:
```
public void DeletePlayerCharacter(string id) {
    var pc = GetPlayerCharacterById(id);

    var userFilter = Builders<User>.Filter.ElemMatch(user => user.Characters, nestedPc => nestedPc.Id == id);
    var userUpdate = Builders<User>.Update.PullFilter(user => user.Characters, nestedPc => nestedPc.Id == id);
    _dataBaseService.userCollection.UpdateOne(userFilter, userUpdate);

    var pcFilter = Builders<PlayerCharacter>.Filter.Eq(pc => pc.Id, id);
    _dataBaseService.playerCharacterCollection.DeleteOne(pcFilter);
}

```

**Review by smashingmouse — COMMENTED** [review]
Looking good, a couple of suggestions, questions

**Pull request merged**